src/base/static/components/input-form/index.js

```diff
@@ -53,25 +52,46 @@ class InputForm extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.isContinuingFormSession) {
-      this.setActiveCategory(
-        this.getCategoryConfig(this.state.selectedCategory)
-      );
+    if (
+      nextProps.isFormResetting ||
+      nextProps.selectedCategoryConfig.category !==
+        this.props.selectedCategoryConfig.category
+    ) {
+      this.initializeForm(nextProps.selectedCategoryConfig);
     }
   }
 
   componentWillUnmount() {
     this.props.map.off("dragend", this.handleDragEnd);
   }
 
-  handleDragEnd() {
-    !this.state.isMapPositioned && this.setState({ isMapPositioned: true });
+  initializeForm(selectedCategoryConfig) {
+    this.isWithCustomGeometry =
+      selectedCategoryConfig.fields.findIndex(
+        field => field.type === constants.MAP_DRAWING_TOOLBAR_TYPENAME
+      ) >= 0;
+    this.attachments = [];
+    this.geometryStyle = null;
+    let fields = ImmutableOrderedMap();
+    selectedCategoryConfig.fields.forEach(field => {
+      fields = fields.set(
+        field.name,
+        ImmutableMap()
+          .set(constants.FIELD_STATE_VALUE_KEY, "")
+          .set(constants.FIELD_RENDER_KEY, Math.random())
```

Comment:
Was this `FIELD_RENDER_KEY` added to force a re-render in the VV form's `onClickContinueForm` handler? If so, do we need it? It seems like setting `this.state.fields` to a new ImmutableMap is sufficient to trigger a re-render.

Reply:
Right-- this exists just to support the VV flavors. Setting a new key forces the old Form components to unmount and new components to mount in their place. This is necessary because we need to run the initialization logic on each Field (to set autofill values, etc.) when the user restarts a VV form. Just re-rendering with a blank value is not enough.

We could change the Field component, however, to perform that initialization logic in the `componentWillReceiveProps` lifecycle method, and pass along a `isRestartingForm` flag or something like that. That seems more involved than just setting a new key, though.